core/src/validator.rs

```diff
@@ -2122,6 +2124,17 @@ fn get_stake_percent_in_gossip(bank: &Bank, cluster_info: &ClusterInfo, log: boo
 fn cleanup_accounts_paths(config: &ValidatorConfig) {
     for accounts_path in &config.account_paths {
         move_and_async_delete_path(accounts_path);
+        if !accounts_path.exists() {
+            // Keep the empty directory
+            create_dir_all(accounts_path).unwrap_or_else(|err| {
+                eprintln!(
+                    "Failed to create accounts directory {}: {}",
+                    accounts_path.display(),
+                    err
+                );
+                exit(1);
+            });
+        }
```

Comment:
When constructing a bank from a snapshot dir, we reproduce the <account_path>/run/* files by hardlinking them from <account_path>/snapshot/\<slot\>/ into /run/.  This hardlinking was failing.  I had to add map_err and more explicit err string to find out that it failed there.  The reason was that at the move_and_async_delete_path step the whole  <account_path>/run directory was deleted.  We need to keep the empty run/ directory so that hardlinking can work.

Reply:
ok, got it.  I will move these debugging changes to a separate PR.

Reply:
moved these debugging changes to https://github.com/solana-labs/solana/pull/30913